### PR TITLE
🐳 Adding Testcontainers example

### PR DESCRIPTION
**Pull request by gunnarmorling**
It's mostly the same as the example from the docs, but fully self-contained and with complete _pom.xml_.

### CODE REVIEW

**Comment by gunnarmorling** [comment]
Pushed one more commit for some (unrelated) change of HTTP -> HTTPS.

File: testcontainers/pom.xml [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    <maven.compiler.source>1.8</maven.compiler.source>
+    <maven.compiler.target>1.8</maven.compiler.target>
+    <version.debezium>1.1.0.Beta2</version.debezium>
+  </properties>
+  <dependencyManagement>
+    <dependencies>
+      <dependency>
+        <groupId>org.testcontainers</groupId>
+        <artifactId>testcontainers-bom</artifactId>
+        <version>1.12.5</version>
```
Comment (jpechane) [inline testcontainers/pom.xml]: Version should go to properties

File: testcontainers/pom.xml [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    <dependency>
+      <groupId>org.postgresql</groupId>
+      <artifactId>postgresql</artifactId>
+      <version>42.2.9</version>
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>org.slf4j</groupId>
+      <artifactId>slf4j-log4j12</artifactId>
+      <version>1.7.28</version>
```
Comment (jpechane) [inline testcontainers/pom.xml]: Would not importing of Debeizum parent set all those versions?

File: testcontainers/src/test/java/io/debezium/examples/testcontainers/DebeziumContainerTest.java [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+public class DebeziumContainerTest {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(DebeziumContainerTest.class);
+
+    private static Network network = Network.newNetwork();
+
+    private static KafkaContainer kafkaContainer = new KafkaContainer()
+            .withNetwork(network);
+
+    public static PostgreSQLContainer<?> postgresContainer = new PostgreSQLContainer<>("debezium/postgres:11")
```
Comment (jpechane) [inline testcontainers/src/test/java/io/debezium/examples/testcontainers/DebeziumContainerTest.java]: Could this be also exteranlized as system property?

File: testcontainers/src/test/java/io/debezium/examples/testcontainers/DebeziumContainerTest.java [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+                        ConsumerConfig.GROUP_ID_CONFIG, "tc-" + UUID.randomUUID(),
+                        ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest"),
+                new StringDeserializer(),
+                new StringDeserializer());
+    }
+
+    private List<ConsumerRecord<String, String>> drain(KafkaConsumer<String, String> consumer, int expectedRecordCount) {
+        List<ConsumerRecord<String, String>> allRecords = new ArrayList<>();
+
+        Unreliables.retryUntilTrue(10, TimeUnit.SECONDS, () -> {
```
Comment (jpechane) [inline testcontainers/src/test/java/io/debezium/examples/testcontainers/DebeziumContainerTest.java]: Why ar enot we using Awaitility?

File: testcontainers/src/test/resources/log4j.properties [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+
+# Root logger option
+log4j.rootLogger=INFO, stdout
+
+# Set up the default logging to be INFO level, then override specific units
+log4j.logger.io.debezium=INFO
+log4j.logger.io.debezium.embedded.EmbeddedEngine$EmbeddedConfig=WARN
+#log4j.logger.io.debezium.embedded.EmbeddedEngine=DEBUG
+log4j.logger.io.debezium.core=DEBUG
+log4j.logger.io.debezium.connector.sqlserver=DEBUG
```
Comment (jpechane) [inline testcontainers/src/test/resources/log4j.properties]: Superfluous logger

File: testcontainers/pom.xml (hunk 1 above)
Comment (gunnarmorling) [reply testcontainers/pom.xml]: I concisiously decedided against it to keep things simple. `version.debezium` is there as its used multiple times, unlike the others.

File: testcontainers/pom.xml (hunk 2 above)
Comment (gunnarmorling) [reply testcontainers/pom.xml]: Interesting; so far I didn't see the parent POM to be used as a BOM as you suggest.

File: testcontainers/src/test/java/io/debezium/examples/testcontainers/DebeziumContainerTest.java (hunk 3 above)
Comment (gunnarmorling) [reply testcontainers/src/test/java/io/debezium/examples/testcontainers/DebeziumContainerTest.java]: Sure, but is it worth it? I'd like to keep things simple.

File: testcontainers/src/test/java/io/debezium/examples/testcontainers/DebeziumContainerTest.java (hunk 4 above)
Comment (gunnarmorling) [reply testcontainers/src/test/java/io/debezium/examples/testcontainers/DebeziumContainerTest.java]: Ok, will change.

File: testcontainers/src/test/resources/log4j.properties (hunk 5 above)
Comment (gunnarmorling) [reply testcontainers/src/test/resources/log4j.properties]: Will remove.

Comment (gunnarmorling) [reply testcontainers/src/test/resources/log4j.properties]: Done.

File: testcontainers/src/test/java/io/debezium/examples/testcontainers/DebeziumContainerTest.java (hunk 4 above)
Comment (gunnarmorling) [reply testcontainers/src/test/java/io/debezium/examples/testcontainers/DebeziumContainerTest.java]: Changed.

**Comment by gunnarmorling** [comment]
Pushed one more commit to address your review remarks, @jpechane.

**Comment by jpechane** [comment]
@gunnarmorling LGTM, when conflicts are resolved you can merge it

**Comment by gunnarmorling** [comment]
> when conflicts are resolved you can merge it

Weird, it seems I somehow merged the original version to master already. This was not my intention, this PR should have initially contributed this. Anyways, I'll take care of it and merge. Thanks for reviewing.

**Comment by gunnarmorling** [comment]
Applied.

**Pull request closed without merging**